docs/getting-started/what-is-dotnet.md

```diff
@@ -1,6 +1,150 @@
+
+- An exception handling mechanism. The runtime, as well as library code, throws exceptions, which you can handle with try/catch statements.   
+
+##The Class Library. 
+
+The class library available in .NET Core is very much like the .NET Framework Class Library for the Windows desktop, with one major difference. Whereas on the Windows desktop is a monolithic library that is part of the operating system and is maintained by Windows Update, on .NET Core it is a modular and factored collection of individual libraries organized by functionality. 
+
+One library, the Microsoft.NetCore.App, is included with the runtime and contains the basic types needed for all application development. Some of these include:
+
+-  Primitives, such as the Boolean type, signed and unsigned integral types, floating-point types, the Char structure.
+  
+-  General-purpose exception classes, such as the ArgumentException, ArgumentNullException, and NullReferenceException types.
+
+- The task types, such as Task and Task<T>, to support asynchronous programming.
+
+- Basic threading types, to support multithreaded application development.
+
+- The Console class, to support the development of console applications. 
+
+Other libraries are installed as [NuGet](http://www.nuget.org/ "NuGet") packages
+
+## .NET Core CLI ##
+
+.NET Core includes a new cross-platform command-line SDK known as the CLI (for Command-Line Interface). The CLI is a UNIX-friendly set of tools for building .NET Core applications. It abstracts the C# compiler, the NuGet package manager, and includes incremental build logic. It can also integrate with the .NET Native tool chain to produce high-performance native apps and libraries. The CLI allows developers to build and test applications with a minimal installation on developer, lab, or production machines. Since some IDEs, such as Visual Studio Code nad Visual Studio, use the CLI under the covers, developers can now choose the level they want to interact with the tool chain by working with the CLI directly, using an editor that uses the CLI, or developing with an IDE that uses the CLI internally.  
+
+## Language Support and Development Environments ##
+
+.NET Core provides a runtime environment for your app, as well as a collection of libraries whose types and members your application can use. .NET Core itself is language agnostic: it is not a programming language, but any programming languages that targets .NET Core can be used to develop .NET Core applications, and an application written in one programming language can seamlessly access types and members in libraries developed by using a different programming language. 
+ 
+You develop and compile .NET Core apps by using the [.NET Compiler Platform](https://github.com/dotnet/roslyn "Roslyn Compiler Platform"), also known as Roslyn, which allows you to write your apps in C# or F#. (At the moment, other languages are not supported, though that will change going forward.) 
+
```

Comment:
@shahid-pk I believe `dotnet` CLI supports only C# and F#. See for example [the documentation for `dotnet new`](https://github.com/dotnet/cli/tree/rel/1.0.0/src/dotnet/commands/dotnet-new#options).